src/lib.rs

```diff
@@ -503,7 +512,8 @@ pub mod pallet {
 			T::Currency::unreserve(&who, reserve);
 
 			NonceCall::<T>::remove(supersig_index);
-			Supersigs::<T>::remove(supersig_index);
+			Members::<T>::remove_prefix(supersig_index, None);
+			TotalMembers::<T>::remove(supersig_index);
 			Calls::<T>::remove_prefix(supersig_index, None);
 			Votes::<T>::remove_prefix(supersig_index, None);
 			UsersVotes::<T>::remove_prefix((supersig_index,), None);
```

Comment:
Just bellow:

```rust
T::Currency::transfer(
				&supersig_id,
				&beneficiary,
				balance,
				ExistenceRequirement::AllowDeath,
			)?;
```
Maybe use `transfer_all` to be more explicit

Reply:
transfer_all is an extrinsic of pallet_balances, and as i use the trait Currency, and i'm not tightly coupled with pallet_balance, i cannot use it